backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java

```diff
@@ -0,0 +1,40 @@
+package com.hackner.musicompass.discogsapi.service;
+
+import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class DiscogsArtistApiService {
+
+    private RestTemplate restTemplate;
+
+    private String baseUrl = "https://api.discogs.com";
+
+    @Autowired
+    public DiscogsArtistApiService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public DiscogsArtistSearchResults getDiscogsArtistByArtistName(String artistName, String accessToken) {
+
+        String url = baseUrl + "/database/search?type=artist&q=" + artistName;
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("User-Agent", "MusiCompass/0.1");
+        headers.add("Authorization", "Discogs token=" + accessToken);
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
```

Comment:
if you intend to include more API calling method in this class you could extract this step into a private helper class avoiding redundant code

Reply:
I decided to include this in the helper class for the entity (as suggested in ur next comment)